packages/admin/cms-admin/src/generator/future/generator.ts

```diff
@@ -39,29 +39,36 @@ type MultiFileFormFieldConfig = { type: "fileUpload"; multiple: true; maxFiles?:
     "maxFileSize" | "readOnly" | "layout" | "accept"
 >;
 
+type InputBaseFieldConfig = {
+    startAdornment?: Adornment;
+    endAdornment?: Adornment;
+};
+
 export type FormFieldConfig<T> = (
-    | { type: "text"; multiline?: boolean }
-    | { type: "number" }
-    | {
+    | ({ type: "text"; multiline?: boolean } & InputBaseFieldConfig)
+    | ({ type: "number" } & InputBaseFieldConfig)
+    | ({
           type: "numberRange";
           minValue: number;
           maxValue: number;
           disableSlider?: boolean;
-      }
+      } & InputBaseFieldConfig)
     | { type: "boolean" }
-    | { type: "date" }
-    | { type: "dateTime" } // TODO add InputBaseFieldConfig once merged (!2645)
-    | {
+    | ({ type: "date" } & InputBaseFieldConfig)
+    | ({ type: "dateTime" } & InputBaseFieldConfig)
+    | ({
           type: "staticSelect";
           values?: Array<{ value: string; label: string } | string>;
           inputType?: "select" | "radio";
-      }
-    | {
+          startAdornment?: Adornment;
+      } & Omit<InputBaseFieldConfig, "endAdornment">)
```

Comment:
This adds `startAdornment` twice

```suggestion
      } & Omit<InputBaseFieldConfig, "endAdornment">)
```